### ISSUE TITLE
Next feature release

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#10484]
That might surprise you:
But I think its wise to go into a new beta phase with current `master` as soon as the lab cache fine tuning is considered final.
`release` and `master` are quickly drifting apart and in general I made good experience with rather small batches of new features.

I do however not know the working state of some other features just implemented, like e.g. the internal Brouter.
Do we have other work in progress on `master` which should either be included or wait?

BTW:
This shall not stop you from including bugfixes into `release`. Those are merged upstream anyhow, and it might still occur, that we are forced to make a bugfix release instead.

**Comment by moving-bits** [comment]
Regarding BRouter integration:
- it's feature complete and working fine AFAICS
- there is one "would be nice" feature missing yet: #10416 (but this one should probably wait for #10310)
- and we can always improve in code and test quality over there (yeah, I love legacy code): #10279, #10291

All of them non-blocking, IMHO.

**Comment by eddiemuc** [comment]
@Lineflyer I assume this means I should rather not merge the new filtering approach to master now?

**Comment by Lineflyer** [comment]
> @Lineflyer I assume this means I should rather not merge the new filtering approach to master now?

Maybe better exactly after next branching?!

**Comment by Lineflyer** [comment]
Before merging branches I think we should include a solution for #10472 and #10477

**Comment by moving-bits** [comment]
> Before merging branches I think we should include a solution for #10472 and #10477

See #10499 and #10500

**Comment by Lineflyer** [comment]
Thanks @moving-bits 

I did however reopen #10477 again as there is still a glitch in this implementation, which will IMHO irritate the basic users. 

If that is fixed we are fine for merging branches IMHO.
Please be a bit conservative until then with merging new features onto `master` as all new stuff would potentially require some testing, finetuning and thus should be merged after the branches have been cut off for a next feature/beta release.

**Comment by moving-bits** [comment]
I've prepared a fix for the missing part of #10477, see #10525.

Besides that PR, I would vote for including #10519 as well (map downloader for "Hylly map" for Finland) - it's a well-defined part within the downloader suite, and should not affect any other areas.

**Comment by Lineflyer** [comment]
@moving-bits Looks good. Thanks for fixing the missing part.
Anything else missing for a beta version?

@baiti The refresh for ALC might also be nice to have, we can still add this while we are in our beta phase. The only difference would be, that your PRs need to target `release` then instead of `master`.
Or is it about to be ready, and we should (for easier workflow) wait a bit to get this merged first?

**Comment by bekuno** [comment]
> Anything else missing for a beta version?

I would finish #10468 first (this week), but retarget this to `release` later would also be ok.

**Comment by baiti** [comment]
> @moving-bits Looks good. Thanks for fixing the missing part.
> Anything else missing for a beta version?
> 
> @baiti The refresh for ALC might also be nice to have, we can still add this while we are in our beta phase. The only difference would be, that your PRs need to target `release` then instead of `master`.
> Or is it about to be ready, and we should (for easier workflow) wait a bit to get this merged first?

From my perspective it is ready to merge right now. It would still be good if the experts could do a quick review. (@fm-sys ? @moving-bits ? how about a quick look at it?)

**Comment by fm-sys** [comment]
IMHO #10531 should and #10532 can be included. 

#10528 is IMHO uncritical and I would vote for merging before the new beta as well...

**Comment by Lineflyer** [comment]
@fm-sys 
#10531 and #10532 are merged now.
For #10528 I will wait for the review result or maybe merge branches before?!

**Comment by baiti** [comment]
> For #10528 I will wait for the review result or maybe merge branches before?!

I have no further changes in mind. So I believe it is up to @fm-sys and @moving-bits to say "Yea or Nay".

**Comment by Lineflyer** [comment]
We do already have the ability to "Set found" for ALC in our code.
AFAICS #10528 will add a refresh capability (which is not that high priority IMHO, as lab caches (without logbook) will potentially not change their content very often.

I vote for merging branches tonight anyway. 
This is no showstopper for #10528 anyhow. If it is considered fit for merge, we can merge it to `release` easily and update the beta version afterwards.

**Comment by baiti** [comment]
> We do already have the ability to "Set found" for ALC in our code.
> AFAICS #10528 will add a refresh capability (which is not that high priority IMHO, as lab caches (without logbook) will potentially not change their content very often.

Without this PR the "Set found" will not work correctly as there is a hidden refresh in the code which is triggered when refreshing lists, in which case the manually set found state of an ALC will be wiped out. This will trigger bug reports.

> 
> I vote for merging branches tonight anyway.
> This is no showstopper for #10528 anyhow. If it is considered fit for merge, we can merge it to `release` easily and update the beta version afterwards.

Ok so go for it ;)

**Comment by baiti** [comment]
> We do already have the ability to "Set found" for ALC in our code.
> AFAICS #10528 will add a refresh capability (which is not that high priority IMHO, as lab caches (without logbook) will potentially not change their content very often.

This actually triggers a point. I need to extract the "IsArchived" state as that can change at least. I will do an additional commit within minutes .....

**Comment by Lineflyer** [comment]
FYA:

`release` has been fast-forwarded to HEAD of `master` in preparation of a new beta version.
Please target bugfixes/completion for merged features to `release` while as new features should be merged to `master`.

Will continue tonight with preparing a beta version.

**Comment by Lineflyer** [comment]
Beta version 2021.05.05-RC has been smoke tested OK and pushed to Google Play Beta channel.

**Comment by baiti** [comment]
> Beta version 2021.05.05-RC has been smoke tested OK and pushed to Google Play Beta channel.

So is: this https://github.com/cgeo/cgeo/pull/10528 now in? I have seen it rebased to release but haven't seen any merge. How can  I tell it's in or not?

**Comment by Lineflyer** [comment]
Its not merged yet, so its not part of that version.

**Comment by fm-sys** [comment]
>So is: this #10528 now in? I have seen it rebased to release but haven't seen any merge. How can I tell it's in or not? 

It's not in, as long as the PR is not yet merged. If it's up to me I would have already merged the PR, but I'm not sure about @moving-bits review comment if it was a change request or just an idea...

**Comment by baiti** [comment]
> > So is: this #10528 now in? I have seen it rebased to release but haven't seen any merge. How can I tell it's in or not?
> 
> It's not in, as long as the PR is not yet merged. If it's up to me I would have already merged the PR, but I'm not sure about @moving-bits review comment if it was a change request or just an idea...

Ahh ok ... it needs to be merged first. I wanted to be sure I understand the process.

I am not in a hurry getting it integrated or out to the road ;)

**Comment by baiti** [comment]
Just be aware that there are bugs in the ALC functionality in the new beta. If ALCs are stored into lists and the user refreshes the caches in the list, the manually set found state is wiped out.

**Comment by Lineflyer** [comment]
@baiti Please open an issue for that (despite its being adressed by the pending PR already).

**Comment by Lineflyer** [comment]
BTW:
RC is available and 33 users currently online with LC connector.
BTW: Does this also count for BM with forecul deactivation?

![image](https://user-images.githubusercontent.com/949669/117219437-8edf4b00-ae05-11eb-85fa-a2883dba2e45.png)

**Comment by baiti** [comment]
can these stats be viewed by regular team members somewhere?

**Comment by fm-sys** [comment]
See https://status.cgeo.org

**Comment by fm-sys** [comment]
> Does this also count for BM with defined deactivation?

Seems like they are counted. Otherwise, there would be more PM's using the LC connector than beta + nightly users which online at all...

BTW, is `master` open for merging new features? 
The current beta is a rather small one, and I think no larger issues are to be expected... (and even if, that's why we have branches ;-))

**Comment by Lineflyer** [comment]
> BTW, is master open for merging new features?

Sure.


> > Does this also count for BM with defined deactivation?
> 
> Seems like they are counted. Otherwise, there would be more PM's using the LC connector than beta + nightly users which online at all...

Thats indeed strange... @moving-bits

**Comment by eddiemuc** [comment]
> > BTW, is master open for merging new features?
> 
> Sure.

As you know I am waiting to merge a rather large change which imho will most likely also start a longer debate and feature tuning: a new filtering model (#10147). Would this be the right time?

**Comment by fm-sys** [comment]
> As you know I am waiting to merge a rather large change which imho will most likely also start a longer debate and feature tuning: a new filtering model (#10147). Would this be the right time?

I would say so. We rarely have as much time as now until another feature release ;-)

**Comment by Lineflyer** [comment]
> As you know I am waiting to merge a rather large change which imho will most likely also start a longer debate and feature tuning: a new filtering model (#10147). Would this be the right time?

Go for it.

**Comment by moving-bits** [comment]
> > > Does this also count for BM with defined deactivation?
> > 
> > Seems like they are counted. Otherwise, there would be more PM's using the LC connector than beta + nightly users which online at all...
> 
> Thats indeed strange... @moving-bits

Hm, from a quick glance everything seems to be fine:
- `StatusUpdater` calls `isActive()` for each connector
- `LCConnector.isActive()` checks for `Settings.isLCConnectorActive()`
- which tests for the same preference key that is touched in the LC connector's settings UI

**Comment by Lineflyer** [comment]
I will push an update to the existing RC to our beta users. We have a few fine tunings of ALC and other bugfixes on `release` since last RC.

**Comment by Lineflyer** [comment]
Updated RC 2021.05.11-RC is live!

**Comment by Lineflyer** [comment]
The Google Play Console and support mail looks fine so far for our beta version.
Things reported by users are already listed here and should probably be fixed before release IMHO:
#10566
#10567
#10568

**Comment by murggel** [comment]
I hope I can create PR for #10598 and a fix for #6711 this evening. 
Would target that to release..

**Comment by Lineflyer** [comment]
My status for release would be the following:
- PRs  #10657 and #10645 should be reviewed and merged first
- Additionally issues #10566, #10567 and #10568 are open bugs with respect to ALC functions. Anyone working on those?

**Comment by Lineflyer** [comment]
Seems we are forced to release due to #10688.
But IMHO we can do that. Remaining things (as listed above) can be done in just another bugfix release.

**Comment by Lineflyer** [comment]
FYI:
I prepared everything for a release. Explicitly I added a link to the ALC related FAQ into the changelog and added #10566 to the FAQ.
I will continue tonight with smoke testing and upload to Google Play.

Please kindly do not merge any open PRs to `release`  and keep `release` unchanged during that phase until release it out.

**Comment by baiti** [comment]
I've been looking into the ALC related filtering bugs but haven't been able to read / understand the code paths yet.

**Comment by Lineflyer** [comment]
Lets go:


- [x] Check/Insert correct version name as headline in ```changelog-release.xml``` 
- [] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid 
- [ ] Trigger notification to status.cgeo.org via CI server once the release is available on Google Play
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

## Release checklist for staged release:

- [ ] Check/Import incoming Translations from Crowdin
- [ ] Check/Insert correct version name as headline in ```changelog-release.xml```
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play to xx% of users (staged release)
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Regularly check Google Play crash reports, support mail, etc. and react accordingly
- [ ] Go/NoGo decision

**Comment by Lineflyer** [comment]
Release 2021.05.20 has been uploaded to Google Play.
So login should be fixed and ALC released.

Waiting now for Google to make it available on Google Play.

**Comment by baiti** [comment]
I have some feedback, don't know where to report: I was using a development build before so I knew I had to do a backup, then deinstall the dev build. By doing so, the cache details folder, in my case some Gigabytes of data got erased, it is not in the backup but worse: I then installed the release from Playstore. Knowing that the wizzard would ask me many many questions, most of which it would find the answers in the backup, at least that is what I hoped, so I bailed out of the wizard, wanted to do a restore from backup. Needless to say  the restore function had no idea where my backup was. I had to traverse through the filesystem and point it to the right location. So far so good. Now I thought it would find all it needs in the backup. No way. I had to answer many many many questions with respect to directory of maps and themes and what not. Twice I was faced with multiple choice questions whether to do something or not to do something (in both cases I said don't do anything). Long story short: It all worked in the and but I think it is way too complicated.

Could this be simplified? Would it be possible to set everything up just as it was when the backup was created, no further questions asked? Point the install to the location where the restore data is and then go and just do it. Remember the answers to those questions I already answered before.

And one more thing: would it be possible to keep that directory with the cache details upon deinstall of the app? It takes me hours to restore those from external storage. 

I am on the virge to completely go to development builds, in which case I won't need to deinstall for upgrades.

Please don't take it as a complaint. My motivation is to sse whether things could potentially be improved

**Comment by Lineflyer** [comment]
Basically you have to follow this: https://manual.cgeo.org/de/replacedevice

There is no way around to reconfirm all SAF directories.
The "do you want to replace" questions would have not been asked if you just set the base directory (to the same as before) and then just restore the backup while in the wizard. Or is there still a bug?

**Comment by baiti** [comment]
Most likely there isn't a bug. It was perhaps just not the smartest idea to abort the wizard. It was actually the first time I tried to be smarter ;)

But what about the data in: `/storage/emultated/0/Androit/data/cgeo.geocaching` ... I believe this is where the gigabytes of cache details go to. Any way to not delete that during a deinstall process?

**Comment by Lineflyer** [comment]
Nope.. But it should not contain vital data for the app to work. Just mainly the log pictures of stored caches and offline logs.

**Comment by baiti** [comment]
you say that's not vital, e.g. not needed to restore? Offline logs I don't need to preserve but I thought for instance spoiler images go there and those I didn't want to loose so far, perhaps I need to rethink my workflow. Would certainly save A LOT of space.

**Comment by Lineflyer** [comment]
Spoiler images go there, you are right. Mainly all pictures and their metadata for each stored caches.

**Comment by eddiemuc** [comment]
Offline log images go there too

**Comment by baiti** [comment]
there is one subdirectory per GC code in there and in each directory there is a handful of files. In th end, with 20,000 cache on the device there are 20,000 directories and let's say 10 files each. Doing manual backup / restore to a server via sftp takes forever. Since we already have a zip / unzup mechanism for the map themes, could access to this data also be run through the zip / unzip? Let's say have one zipped file per GC code and eliminate the many directories and the many files. I'd guess this speeds up backup / restore significantly and is not much of a burden in cgeo at runtime when these files need to be accessed.
